cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/ObjectMappedTableProperties.java

```diff
@@ -40,22 +41,53 @@
   /**
    * The type of object in the table.
    */
-  public static final String TYPE = "type";
+  public static final String OBJECT_TYPE = "object.type";
 
   /**
    * The schema of objects in the table.
    */
-  public static final String SCHEMA = "schema";
+  public static final String OBJECT_SCHEMA = "object.schema";
+
+  /**
+   * The name of the key to use when exploring the table. Defaults to "key".
+   */
+  public static final String EXPLORE_KEY_NAME = "explore.key.name";
+
+  /**
+   * The type of the key when exploring the table. Defaults to bytes.
+   */
+  public static final String EXPLORE_KEY_TYPE = "explore.key.type";
```

Comment:
for the third, maybe we could add a way, but it would not actually be enforced by the Java API, so I think it is better not to do that unless there is a corresponding way to enforce it in the Java API.